### ISSUE TITLE
Unable to progress forever if leader crashes during commit

### ISSUE THREAD

**Issue by noirgif** [tikv/tikv#6224]
### What version of TiKV are you using?

4.0.0-alpha

### Steps to reproduce

1. Setup a 3-node TiKV cluster, wait for the election to take place
2. After a leader is elected, initiate a command(I used Get() in the goclient)
3. Inject an I/O error or out-of-space error to the RocksDB when it is appending to the state log(`db/000015.log` in this case)

### What did you expect?

The cluster should be able to progress.

### What did happened?

Current term is 5.

1. The leader(tikv2) crashed with the error(the fatal error itself is not shown in the log for unknown reason, but there's two 'Welcome to TiKV')
2. The crashed leader recovered
3. All become followers for Term 6.
4. New election and tikv2 is elected as leader of Term 7.
4. New reads are redirected from tikv0 and tikv1 to tikv2, and fail at tikv2, because they are 'rejected by term check'. The cluster is unable to make future progress.

My thought is that the leader neither committed its decision to the disk, nor aborted the proposal, which made the cluster blocked.

Log files:
[tikv0.log](https://github.com/tikv/tikv/files/3951449/tikv0.log)
[tikv1.log](https://github.com/tikv/tikv/files/3951450/tikv1.log)
[tikv2.log](https://github.com/tikv/tikv/files/3951451/tikv2.log)
[client.log](https://github.com/tikv/tikv/files/3951505/client.log)

**Comment by zhangjinpeng1987** [comment]
@BusyJay @hicqu PTAL

**Comment by BusyJay** [comment]
The cluster will not always block. When the leader can't commit its decision, it will crash. So one of remaining followers will become leader and then the cluster is recovered.

There are two "Welcome" in tikv2's log, I think that's because leader panics due to IO error. After second "Welcome", tikv2 is not leader anymore. I think you can wait longer and see what happen.

I think there is a problem with master that stacktrace can be lost during panic. Can you use the latest master and try again? So we can see what actually triggers restarting.

**Comment by noirgif** [comment]
Yeah, it seems the cluster recovered when I restart them again. I'm not sure why it cannot recover during the request but I will close this issue for now.

The restarting should be triggered by the docker-compose script in tikv/tidb-docker-compose, which restarts the container on failure.

**Comment by BusyJay** [comment]
To get a quicker election, you can modify the election timeout tick to a smaller value. Current default is 10 seconds, so the maximum election timeout is 20 seconds.

**Comment by noirgif** [comment]
Thank you! I see it has a maximum of 20s, and in the log the timeout has been set to 18 ticks.

However, tikv1 set the election timeout to be 10 ticks at 12:28:37.761, right before the leader(tikv2) fails, and it started a new election not until 12:29:04.657. After that it set the election timeout back to 18 ticks at 12:29:04.657, when it became the leader.

**Comment by noirgif** [comment]
Issue #1569 may be related to this problem. How can I measure how long a tick actually takes?
